README.md

```diff
@@ -0,0 +1,486 @@
+# sanctuary-def
+
+sanctuary-def facilitates the definition of curried JavaScript functions which
+are explicit about the number of arguments to which they may be applied and the
+types of those arguments.
+
+```javascript
+const $ = require('sanctuary-def');
+
+const def = $.create($.env);  // Explained in "Environments" section
+
+//    add :: Number -> Number -> Number
+const add =
+def('add', {}, [$.Number, $.Number, $.Number], (x, y) => x + y);
+```
+
+Applying `add` to two arguments gives the expected result:
+
+```javascript
+add(2, 2);
+// => 4
+```
+
+Applying `add` to greater than two arguments results in an exception being
+thrown:
+
+```javascript
+add(2, 2, 2);
+// ! TypeError: ‘add’ requires two arguments; received three arguments
+```
+
+Applying `add` to fewer than two arguments results in a function awaiting the
+remaining arguments. This is known as partial application. Partial application
+is convenient as it allows more specific functions to be defined in terms of
+more general ones:
+
+```javascript
+//    inc :: Number -> Number
+const inc = add(1);
+
+inc(7);
+// => 8
+```
+
```

Comment:
Excellent point, @boxofrox. Since none of us has much experience with the module at this point, it's too early to state a preference for one approach or the other. It's deserves to be mentioned in the README though. We could insert the following:

> It's worth noting that defining functions via partial application reduces the helpfulness of error messages:
>
> ```javascript
> inc('0');
> // ! TypeError: ‘add’ expected a value of type Number as its second argument; received "0"
> ```
>
> This can be avoided by using `def` to define the specialized function:
>
> ```javascript
> //    inc :: Number -> Number
> const inc = def('inc', {}, [$.Number, $.Number], add(1));
> ```